sgkit/distance/api.py

```diff
@@ -0,0 +1,102 @@
+import dask.array as da
+import numpy as np
+
+from sgkit.distance import metrics
+from sgkit.typing import ArrayLike
+
+
+def pairwise_distance(
+    x: ArrayLike,
+    metric: str = "euclidean",
+) -> np.ndarray:
+    """Calculates the pairwise distance between all pairs of vectors in the
```

Comment:
```suggestion
    """Calculates the pairwise distance between all pairs of row vectors in the
```

Just to help make obvious this is computing distance between **rows**. (And thus when computing distance between samples, user will have to transpose input.)